scripts/memory_logger.py

```diff
@@ -1,110 +1,161 @@
+to stdout, or stderr when displaying the prompt to the user (informing the user, the debugger
+is ready to receive input). However, this class may someday be able to, which is why
+the self.debugger variable is present.
+  """
+  def __init__(self, arguments):
+    self.last_position = 0
+    self.debugger = arguments.debugger[0]
+
   def _parseStackTrace(self, gibberish):
-    return gibberish
-
-  def _run_commands(self, commands):
-    tmp_text = ''
-    return_obj = lldb.SBCommandReturnObject()
-    for command in commands:
-      self.command_interpreter.HandleCommand(command, return_obj)
-      if return_obj.Succeeded():
-        if command == 'process status':
-          tmp_text += '\n########################################################\n## Process Status:\n##\n'
-          tmp_text += return_obj.GetOutput()
-        elif command == 'bt':
-          tmp_text += '\n########################################################\n## Backtrace:\n##\n'
-          tmp_text += return_obj.GetOutput()
-    return tmp_text
-
-  def getStackTrace(self, pid):
-    event = lldb.SBEvent()
-    lldb_results = ''
-    state = 0
-    attach_info = lldb.SBAttachInfo(int(pid))
-    process = self.target.Attach(attach_info, self.error)
-    process.GetBroadcaster().AddListener(self.listener, lldb.SBProcess.eBroadcastBitStateChanged)
-    done = False
-    while not done:
-      if self.listener.WaitForEvent(lldb.UINT32_MAX, event):
-        state = lldb.SBProcess.GetStateFromEvent(event)
-        if state == lldb.eStateExited:
-          done = True
-        elif state == lldb.eStateStopped:
-          lldb_results = self._run_commands(['process status', 'bt', 'cont'])
-          done = True
-        elif state == lldb.eStateRunning:
-          self._run_commands(['process interrupt'])
-      if state == lldb.eStateCrashed or state == lldb.eStateInvalid or state == lldb.eStateExited:
-        return 'Binary exited before sample could be taken'
-      time.sleep(0.03)
-
-    # Due to some strange race condition we have to wait until eState is running
-    # before we can pass the 'detach, quit' command. Why we can not do this all in
-    # one go... bug?
-    done = False
-    while not done:
-      if self.listener.WaitForEvent(lldb.UINT32_MAX, event):
-        state = lldb.SBProcess.GetStateFromEvent(event)
-        if state == lldb.eStateRunning:
-          self._run_commands(['detach', 'quit'])
-          done = True
-      if state == lldb.eStateCrashed or state == lldb.eStateInvalid or state == lldb.eStateExited:
-        return 'Binary exited before sample could be taken'
-      time.sleep(0.03)
-    return self._parseStackTrace(lldb_results)
-
-class GDB:
-  def _parseStackTrace(self, gibberish):
-    not_gibberish = re.findall(r'\(gdb\) (#.*)\(gdb\)', gibberish, re.DOTALL)
+    not_gibberish = re.findall(r'\(' + self.debugger + '\) (#.*)\(' + self.debugger + '\)', gibberish, re.DOTALL)
```

Comment:
Here's another one of those DOTALLs. These can be problematic if you have a long string (several lines) and you match something on an early line then several lines followed by the remaining part of your expression on a lower line.

The bottom line is if you don't need to match across lines, don't use DOTALL.